### ISSUE TITLE
CVEs display is not responsive

### ISSUE THREAD

**Issue by anthonydillon** [canonical-web-and-design/ubuntu.com#9325]
This view is not useful:
![Screenshot_20210227-020932.jpg](https://user-images.githubusercontent.com/1413534/109372361-02faf100-78a1-11eb-8b12-38c2016f8d7f.jpg)

**Comment by SirSamTumless** [comment]
@elioqoshi I am moving this to next iteration.

**Comment by SirSamTumless** [comment]
@elioqoshi Is this something you think you can work on in the next two weeks?

**Comment by elioqoshi** [comment]
@SirSamTumless I don't think I can finish it but might be able to start work on this next week.

**Comment by elioqoshi** [comment]
@SirSamTumless I think this is dependent on Vanilla table work but should be looked at probably at some point

**Comment by anthonydillon** [comment]
What Vanilla work is this dependent on? Happy to expedite it.

**Comment by elioqoshi** [comment]
@anthonydillon I don't think this is the whole story but sticky table headings is likely a part of it https://github.com/canonical-web-and-design/vanilla-framework/issues/3434

@lyubomir-popov might know more in detail what dependency there is?

**Comment by lyubomir-popov** [comment]
@elioqoshi @anthonydillon I don't think this is blocked on #3434. That issue is about sticky header row, which might be a nice to have here, but wouldn't really sddress the fact that everything gets truncated. What is need here is to pick from the different responsive strategies outlined in the [modular table spec](https://discourse.ubuntu.com/t/responsive-behaviours/19855) and implement it.

**Comment by anthonydillon** [comment]
Could we update the display to cards? As the versions grow this UI will not scale.

**Comment by spencerbygraves** [comment]
@anthonydillon @elioqoshi something like this? I've tried an accordion when there is more than one package, not sure if that will work though.

![image](https://user-images.githubusercontent.com/9105999/142631375-39fdf3c4-d5e0-4f78-944b-c44c2553371a.png)

**Comment by anthonydillon** [comment]
This looks good. The accordion icon is different from the [basic one](https://vanillaframework.io/docs/patterns/accordion)?

I would hope the table card pattern could produce this.

**Comment by spencerbygraves** [comment]
>The accordion icon is different from the basic one?

@anthonydillon my mistake.

**Comment by SaidMarar** [comment]
What about a sticky table with first column  and horizontal scroll ?
![sticky-table](https://user-images.githubusercontent.com/20561190/142702790-0284908b-21fa-4c11-b962-274e7a121e16.gif)

**Comment by anthonydillon** [comment]
@SaidMarar I like it! Would be much easier to implement. A draw back to this approach is that it's not apparent that you can scroll the content horizontally. I wonder if there is something that could be added to make that clear?

**Comment by davegoddard42** [comment]
I like this as a solution, but I'm curious as to how this would feel on mobile. It's being tested here on a desktop browser with the inspect tool. Will the drag gesture interfere with forward/back gesture for navigating forward/back in the browser?

Also, it doesn't appear to be immediately obvious to the user that the table has this interaction. Would a user realise they can drag left/right on the table?

If we want to purse this, I'd suggest adding a vertical line between the swipeable and non swipeable columns. Also, use some thing to show it's possible to swipe - I've seen something like the image below used in the past for this.

![image](https://user-images.githubusercontent.com/25453620/142841266-8f1d4af5-9665-465b-bc4c-55db644a0074.png)

Happy to explore designs on this.

**Comment by davegoddard42** [comment]
Ah, I just saw your comment @spencerbygraves!

**Comment by spencerbygraves** [comment]
@anthonydillon let's try the scrolling table approach then. I don't think we have a pattern to make the horizontal scroll more obvious, but adding a vertical line or slight shadow as @davegoddard42 suggests would be a good start.

Thanks.

**Comment by davegoddard42** [comment]
@spencerbygraves @anthonydillon 

I'm happy to explore some designs that make it more obvious - what is the timescale on this?

I think it's important we test any solution on a mobile device to get a feel for how it works on the target device too.

**Comment by anthonydillon** [comment]
Looking at the analytics there is only a very small amount of visitors to that page (0.8% site traffic) and much less on small screens. Therefore I would have to say unfortunately it would be low priority if you are busy at the moment.

**Comment by davegoddard42** [comment]
I could take a look next iteration. We could close this issue and I'll create a new one for CVEs Display Improvements?

**Comment by anthonydillon** [comment]
Thanks Dave. This issue is for CVS display improvements. Don't think closing and opening a new one is necessary.

**Comment by elioqoshi** [comment]
Just a note to consider accessibility in terms of table horizontal scrolling. Tables in general are bad accessibility-wise already. Maybe we could offer various ways to display CVE's? People might be looking for a specific CVE or any CVE affecting their OS version or any CVE with high severity so creating a more flexible experience here could go a long way (but that's also a bigger commitment). What has already been suggested would improve the current implementation already

**Comment by sowasred2012** [comment]
Reopening - this issue is valid again, see https://github.com/canonical-web-and-design/ubuntu.com/pull/10975

**Comment by SaidMarar** [comment]
The fix for responsive table should be available only in mobile and not in desktop. to avoid this, we should use **media queries** to apply sticky on the first column. also be aware about nested columns.
